File: 박찬영/10866.js
const fs = require("fs");
const filePath =
  process.platform === "linux" ? "/dev/stdin" : "./박찬영/input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");

console.log(input);
// 자료구조 : 스택, 큐

class Oper {
  constructor() {
    this.arr = [];
  }

  // 정수 X를 덱의 앞에 넣는다.
  push_front(x) {
    this.arr.unshift(x);
  }

  // 정수 X를 덱의 뒤에 넣는다.
  push_back(x) {
    this.arr.push(x);
  }

  // 덱의 가장 앞에 있는 수를 빼고, 그 수를 출력한다. 만약, 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
  pop_front() {
    if (this.arr.length == 0) return -1;
    return this.arr.shift();
  }

  // 덱의 가장 뒤에 있는 수를 빼고, 그 수를 출력한다. 만약, 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
  pop_back() {
    if (this.arr.length == 0) return -1;
    return this.arr.pop();
  }

  // 덱에 들어있는 정수의 개수를 출력한다.
  size() {
    return this.arr.length;
  }

  // 덱이 비어있으면 1을, 아니면 0을 출력한다.
  empty() {
    return this.arr.length == 0 ? 1 : 0;
  }

  // 덱의 가장 앞에 있는 정수를 출력한다. 만약 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
  front() {
    if (this.arr.length == 0) return -1;
    return this.arr[0];
  }

  // 덱의 가장 뒤에 있는 정수를 출력한다. 만약 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
  back() {
    if (this.arr.length == 0) return -1;
    return this.arr.at(-1);
  }
}

const n = +input.shift();
const oper = new Oper();
let result = [];

for (let i = 0; i < input.length; i++) {
  const [명령, 숫자] = input[i].split(" ");

  if (숫자 === undefined) {
    let res = oper[명령]();
    result.push(res);
  } else {
    oper[명령](숫자);
  }
}

console.log(result.join("\n"));

File: 이승현/10866.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int length = scanner.nextInt();
		Deque deque = new Deque();

		for (int i = 0; i < length+1; i++) {
			String[] command = scanner.nextLine().split(" ");
			switch (command[0]) {
			case "push_front":
				deque.push_front(Integer.parseInt(command[1]));
				break;
			case "push_back":
				deque.push_back(Integer.parseInt(command[1]));
				break;
			case "pop_front":
				System.out.println(deque.pop_front());
				break;
			case "pop_back":
				System.out.println(deque.pop_back());
				break;
			case "front":
				System.out.println(deque.front());
				break;
			case "back":
				System.out.println(deque.back());
				break;
			case "size":
				System.out.println(deque.size());
				break;
			case "empty":
				System.out.println(deque.empty());
				break;
			}
		}
	}
}

class Deque {
	private int[] arr;

	public Deque() { // 생성자 배열 선언
		this.arr = new int[0];
	}

	public void push_back(int input) { //뒤에 추가
		int[] temp = new int[this.arr.length + 1]; // 기존 배열 한칸 늘린거 선언
		for (int i = 0; i < this.arr.length; i++) temp[i] = arr[i]; // 배열 복사
		temp[temp.length - 1] = input; // 입력값을 마지막에 추가
		this.arr = temp; // 배열 저장
	}

	public void push_front(int input) { //앞에 추가
		int[] temp = new int[this.arr.length + 1]; // 기존 배열 한칸 늘린거 선언
		temp[0] = input; // 입력값을 앞에 추가
		for (int i = 0; i < this.arr.length; i++) temp[i + 1] = arr[i]; // 기존 내용 복사
		this.arr = temp; // 배열 저장
	}

	public int pop_back() {
		if (arr.length == 0) return -1; // 배열 크기 0이면 -1 반환
		int back = arr[arr.length - 1]; // 출력값은 맨 뒤에 선택
		int[] temp = new int[this.arr.length - 1]; // 배열 한칸 줄인거 선언
		for (int i = 0; i < temp.length; i++) temp[i] = arr[i]; // 기존 내용 복사
		this.arr = temp; // 배열 저장
		return back; // 결과 반환

	}

	public int pop_front() {
		if (arr.length == 0) return -1; // 배열 크기 0이면 -1 반환
		int front = arr[0]; // 출력값은 맨 앞에 선택
		int[] temp = new int[this.arr.length - 1]; // 배열 한칸 줄인거 선언
		for (int i = 0; i < temp.length; i++) temp[i] = arr[i + 1]; // 기존 내용 복사
		this.arr = temp; // 배열 저장
		return front; // 결과 반환
	}

	public int front() {
		if (arr.length == 0) return -1; // 배열 크기 0이면 -1 반환
		return arr[0];
	}

	public int back() {
		if (arr.length == 0) return -1; // 배열 크기 0이면 -1 반환
		return arr[arr.length - 1];
	}

	public int size() {
		return arr.length;
	}

	public int empty() {
		return (arr.length > 0) ? 0 : 1;
	}
}
